flang/test/Lower/OpenACC/acc-loop.f90

```diff
@@ -18,6 +22,217 @@ program acc_loop
 !CHECK:      acc.loop {
 !CHECK:        fir.do_loop
 !CHECK:        acc.yield
+!CHECK-NEXT: }
+
+ !$acc loop seq
+  DO i = 1, n
+    a(i) = b(i)
+  END DO
+
+!CHECK:      acc.loop {
+!CHECK:        fir.do_loop
+!CHECK:        acc.yield
+!CHECK-NEXT: } attributes {seq}
+
+  !$acc loop auto
+  DO i = 1, n
+    a(i) = b(i)
+  END DO
+
+!CHECK:      acc.loop {
+!CHECK:        fir.do_loop
+!CHECK:        acc.yield
+!CHECK-NEXT: } attributes {auto}
+
+  !$acc loop independent
+  DO i = 1, n
+    a(i) = b(i)
+  END DO
+
+!CHECK:      acc.loop {
+!CHECK:        fir.do_loop
+!CHECK:        acc.yield
+!CHECK-NEXT: } attributes {independent}
+
+  !$acc loop gang
+  DO i = 1, n
+    a(i) = b(i)
+  END DO
+
+!CHECK:      acc.loop gang {
+!CHECK:        fir.do_loop
+!CHECK:        acc.yield
+!CHECK-NEXT: }
+
+  !$acc loop gang(num: 8)
+  DO i = 1, n
+    a(i) = b(i)
+  END DO
+
+!CHECK:      [[GANGNUM1:%.*]] = constant 8 : i32
+!CHECK-NEXT: acc.loop gang(num: [[GANGNUM1]]) {
+!CHECK:        fir.do_loop
+!CHECK:        acc.yield
+!CHECK-NEXT: }
+
+  !$acc loop gang(num: gangNum)
+  DO i = 1, n
+    a(i) = b(i)
+  END DO
+
+!CHECK:      [[GANGNUM2:%.*]] = fir.load %{{.*}} : !fir.ref<i32>
+!CHECK-NEXT: acc.loop gang(num: [[GANGNUM2]]) {
+!CHECK:        fir.do_loop
+!CHECK:        acc.yield
+!CHECK-NEXT: }
```

Comment:
Should any of these be checking that there are _not_ `attributes` dicts tagged on the end? (A partial match would succeed either way.)